modules/bibupload/lib/bibupload.py

```diff
@@ -1995,11 +1996,8 @@ def _process_document_moreinfos(more_infos, docname, version, docformat, mode):
                             guessed_format = guess_via_magic(downloaded_url)
                             if guessed_format != docformat:
                                 raise RuntimeError("Given URL %s was supposed to refer to format %s but was found to be of format %s. Is this document behind an authentication page?" % (url, docformat, guessed_format))
-                            from PyPDF2 import PdfFileReader
-                            from PyPDF2.utils import PdfReadError
-                            try:
-                                PdfFileReader(open(downloaded_url))
-                            except PdfReadError as err:
+                            exit_code, dummy_out, err = run_shell_command("pdfinfo", downloaded_url)
```

Comment:
I think this needs to be `run_shell_command("pdfinfo %s", downloaded_url)`
let's hope that the implicit `escape_shell_arg` takes care of filenames with whitespace and special shell character
also, it assumes pdfinfo is on the PATH

Reply:
Yes, `escape_shell_arg` does that. Thanks for spotting the missing pattern.